### PR TITLE
[makefile] add helper functions 

### PR DESCRIPTION
**Pull request by nammn**
### What does this PR do?

Adding helper output to the makefile.

Please Note: 
- that make now defaults to help and one can choose a command from there. 
- I just added some comments and sections because I don't know all of them. Feel free to add some additionally later. 
```
⛵ gke_datadog-sandbox_europe-west1-b_nam-test (datadog) in datadog-operator on  nnguyen/add-make-help via 🐹 v1.15.3 took 4s 
❯ make
Usage: make <OPTIONS> ... <TARGETS>

Available targets are:


Usage:
  make <target>

Development
  all                        Build test
  build                      Builds manager + kubectl plugin
  fmt                        Run go fmt against code
  vet                        Run go vet against code

Deploy
  manager                    Build manager binary
  run                        Run against the configured Kubernetes cluster in ~/.kube/config
  install                    Install CRDs into a cluster
  uninstall                  Uninstall CRDs from a cluster
  deploy                     Deploy controller in the configured Kubernetes cluster in ~/.kube/config
  manifests                  Generate manifestcd s e.g. CRD, RBAC etc.
  generate                   Generate code
  docker-build               Build the docker image
  docker-push                Push the docker image

Test
  test                       Run tests
  controller-gen             Find or download controller-gen, download controller-gen if necessary
  bundle                     Generate bundle manifests and metadata, then validate generated files.
  bundle-build               Build the bundle image.

Datadog Custom part
  patch-crds                 Patch-crds
  lint                       Lint
  verify-license             Verify licenses
  tidy                       Run go tidy
  vendor                     Run go vendor
  help                       Show this help screen.


```
### Motivation

Outdated make docs

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Review by ahmed-mez — COMMENTED** [review]
Thanks @nammn ! could you update https://github.com/DataDog/datadog-operator/blob/master/docs/how-to-contribute.md as well?

File: Makefile [hunk 1]
```diff
@@ -30,69 +30,69 @@ else
...
 
-all: build test
+##@ Development
 
-build: manager kubectl-datadog
+all: build test ## Build test
 
-# Run tests
-test: build manifests verify-license gotest
+build: manager kubectl-datadog ## build
```
Comment (vboulineau) [inline Makefile:37]: Maybe state that it builds manager + kubectl plugin

File: Makefile [hunk 2]
```diff
@@ -30,69 +30,69 @@ else
...
-docker-push:
+docker-push: ## Push the docker image
 	docker push ${IMG}
 
-# find or download controller-gen
-# download controller-gen if necessary
-controller-gen: install-tools
+##@ Test
+
+test: build manifests verify-license gotest ## Run tests
```
Comment (vboulineau) [inline Makefile:89]: We can be more precise: Run unit tests and E2E tests

**Review by vboulineau — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

File: docs/how-to-contribute.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 This project uses the `go module`. Be sure to have it activated with: `export GO111MODULE=on`.
 
+to list all the available make commands run:
```
Comment (kayayarai) [inline docs/how-to-contribute.md:5]:
```suggestion
To list the available `make` commands, run:
```

**Review by kayayarai — APPROVED** [review]
Minor edits to improve readability

**Pull request merged**